### ISSUE TITLE
perf(training): add configurable DataLoader worker and pinned-memory path for predictive planner

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6488]
Parent issue: #6482

```yaml
archetype: implementation
evidence_tier: smoke
linked_policy:
  - docs/dev_guide.md
```

## Goal

Remove the first bounded data-transfer bottleneck from the predictive-planner custom training loop without coupling it to automatic mixed precision or replay-buffer changes.

## Scope

- Add explicit config/CLI settings for DataLoader worker count and pinned memory in `train_predictive_planner.py` and its canonical config surface.
- Use non-blocking device transfer only when pinned-memory/device conditions make it valid.
- Preserve safe CPU defaults and deterministic seed/worker behavior.
- Record effective loader settings in the training manifest.
- Add focused tests for CPU defaults, configured workers, pinned-memory gating, and manifest provenance without requiring a GPU.

## Non-goals

- No automatic mixed precision.
- No replay-buffer implementation change.
- No batch-size, optimizer, model, dataset, or training-objective change.
- No performance or quality claim without a separate measured run.

## Definition of done

- [ ] Loader settings are explicit, validated, and recorded.
- [ ] CPU-only execution remains supported and deterministic.
- [ ] Non-blocking transfers are enabled only on the intended pinned-memory device path.
- [ ] Focused tests pass without GPU hardware.

## Validation / Testing

- Focused predictive-planner training/config tests.
- One CPU smoke with `num_workers=0`.
- `uv run ruff check` and `uv run ruff format --check` on changed files.
- `git diff --check`.

## Blocked by

None.

## Estimates

Best estimate: 4 hours. Complexity: low-medium. Confidence: 0.9.

**Comment by ll7** [comment]
Draft PR opened for independent verification: https://github.com/ll7/robot_sf_ll7/pull/6518

Implements the #6488 contract: configurable `--num-workers` / `--pin-memory` (+ `--persistent-workers` / `--prefetch-factor`) for the predictive-planner training loop, with safe defaults that preserve the current single-process CPU path. `non_blocking=True` is gated on both pinned memory **and** CUDA (stays blocking/deterministic on CPU); a deterministic seeded `worker_init_fn` + generator attach only when `num_workers > 0`; effective loader settings are recorded in `training_summary.json` under a new `data_loader` key.

Scope kept tight: no AMP, no replay-buffer change (`train_distributional_rl.py` / `robot_sf/training` untouched), no batch-size/optimizer/model/dataset/training-objective change. Typed `feat` (not `perf`) on purpose — this PR only adds the configurable path and makes **no** performance claim; a separate measured run is required before any speed-up claim. Left as `Refs #6488` (not `Closes`) so the issue stays open until verified.

Validation (CPU-only, all green): `ruff check`, `ruff format --check`, `git diff --check`, and both `tests/training/test_train_predictive_planner_loader_settings.py` (14 new) and the existing `test_train_predictive_planner_dataset_diagnostics.py` (16).

**Comment by ll7** [comment]
Issue-audit 2026-08-11: closing as completed; PR #6518 merged on 2026-08-01 and delivers the configurable DataLoader worker and pinned-memory path.
